ggml/CMakeLists.txt

```diff
@@ -238,12 +239,15 @@ set_target_properties(ggml PROPERTIES PUBLIC_HEADER "${GGML_PUBLIC_HEADERS}")
 install(TARGETS ggml PUBLIC_HEADER)
 
 if (BUILD_SHARED_LIBS)
-    install(TARGETS ggml LIBRARY)
+    install(TARGETS ggml      LIBRARY)
+    install(TARGETS ggml-base LIBRARY)
 endif()
 
+# FIXME: this should be done in the backend cmake files
 if (GGML_METAL)
+    # FIXME: does this need to be installed with GGML_METAL_EMBED_LIBRARY?
     install(
-        FILES src/ggml-metal.metal
+        FILES src/ggml-metal/ggml-metal.metal
```

Comment:
The `ggml-metal.metal` should not be needed to install when `GGML_METAL_EMBED_LIBRARY` is defined.

Should we move this section to `ggml-metal/CMakeLists.txt` or something prevents from doing so?

Reply:
I believe it can be moved, but I am bit fuzzy about the install stuff in cmake, so I am not exactly sure.

Reply:
Seems to work OK, but I'm not 100% sure either. Let's leave like this for now and resolve later.